MIGRATING.md

```diff
@@ -4,6 +4,119 @@ This guide describes the steps to upgrade dd-trace from a major version to the
 next. If you are having any issues related to migrating, please feel free to
 open an issue or contact our [support](https://www.datadoghq.com/support/) team.
 
+## 2.0 to 3.0
+
+### Node 12 is no longer supported
+
+Node 12 has been EOL since April 2022 and is no longer supported. Generally
+speaking, we highly recommend always keeping Node up to date regardless of our
+support policy.
+
+### HTTP query string reported by default
+
+HTTP query strings are now reported by default as part of the `http.url` tag.
+This change is considered breaking only because there might be sensitive data
+in the query string. A default regular expression based obfuscator is provided
+for common use cases like API keys, but if your use case is not covered, the
+[DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP](https://datadoghq.dev/dd-trace-js/#tracer-settings)
+environmnet variable can be used to control what is obfuscated, and a value of
+`.*` would redact the query string entirely.
+
+### HTTP operation name change
+
+The HTTP integration now uses `web.request` for incoming requests and continues
+to use `http.request` for outgoing requests. When using a supported Web
+framework like Express, this change will have no effect because the root span
+would already have an operation name override like `express.request`.
+Any [monitor](https://docs.datadoghq.com/monitors/create/types/apm/?tab=apmmetrics)
+on `http.request` for incoming requests should be updated to `web.request`.
+
+With this change, both operation names also appear under the main service name
+and are no longer split between the server service name and a separate client
+service name suffixed with `-http-client`.
+
+### gRPC operation name change
+
+The gRPC integration now uses `grpc.server` for incoming requests and
+`grpc.client` for outgoing requests. Any
+[monitor](https://docs.datadoghq.com/monitors/create/types/apm/?tab=apmmetrics)
+on `grpc.request` should be updated to one of these.
+
+With this change, both operation names also appear under the main service name
+and are no longer split between the server service name and a separate client
+service name suffixed with `-http-client`.
+
+### Removal of `fs` integration
+
+The `fs` integration was removed as it was originally added without an actual
+use case, and it's been problematic ever since it's noisy, the output is
```

Comment:
agreed. Maybe just 
```
The `fs` integration was removed as it was originally added without an actual
use case, and it's been problematic ever since. It's noisy ...
```

Reply:
That was actually the intent, fixing.